### DISCUSSION TITLE
Usage with URQL

### DISCUSSION THREAD

**Discussion by vaniyokk** [0no-co/graphql.web#20] [General]
@kitten Sorry for the dummy question, maybe I misunderstood something.

I'm using `urql` in my Next.js application and want to reduce my bundle using this package.
I noted that i'm still loading both `graphql` and `graphql.web`

![12096](https://github.com/0no-co/graphql.web/assets/1783210/fc142cc9-38a6-40d8-b1bb-5623f190dfcc)

I followed a guide to setup webpack alias for `graphql` package in client bundle. After that, I started receiving following errors for all of my client-side queries.

![72689](https://github.com/0no-co/graphql.web/assets/1783210/92c22277-3d48-4b0f-852a-bade5ce9c48b)

The files generated by `graphql-codegen` contains imports of gql utility from `graphql-tag`, which depends on `graphql` and seems like it's not working with a `graphql.web` replacement.

What I'm doing wrong?

**Comment by kitten** [comment]
As per the readme, there's an alias package and instructions and how to use it here: https://github.com/0no-co/graphql-web-lite

Also, you'll want to point your GraphQL Code Generator at `@urql/core`‘s version of `gql` or rather use that directly instead of `graphql-tag` to avoid that dependency entirely, or use the generator to precompile your queries to the AST representation of your queries

**Comment by vaniyokk** [comment]
@kitten thanks for response!

I've read the instructions carefully and implemented the Next.js guide. Just tried to point the `gqlImport` config of `@graphql-codegen/typescript-urql` to `@urql/core` and seems like the client-side issue now fixed and bundle contains only `graphql.web`.

I'm already using `@graphql-codegen/typescript-urql` generator for queries. Is this one you mentioned or there is better one?

BTW, in the `@graphql-codegen/typescript-urql` package Docs there is a following note
![37339](https://github.com/0no-co/graphql.web/assets/1783210/316b7599-7c05-4da9-9285-b9ef93c28250)

Is it that better?
